### PR TITLE
Clean up the dye color processors and add support for beds.

### PR DESCRIPTION
**Pull request by JBYoshi**
- Replace the multi-target `DataProcessor` with individually-targeted dual processors.
- Implement `DyeableData` for beds. See SpongePowered/SpongeAPI#1700.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/data/DataRegistrar.java [hunk 1]
```diff
@@ -762,9 +770,7 @@ public static void setupSerialization() {
         DataUtil.registerValueProcessor(Keys.VANISH, new VanishValueProcessor());
         DataUtil.registerValueProcessor(Keys.VANISH_IGNORES_COLLISION, new VanishCollisionValueProcessor());
         DataUtil.registerValueProcessor(Keys.VANISH_PREVENTS_TARGETING, new VanishTargetValueProcessor());
-        DataUtil.registerValueProcessor(Keys.DYE_COLOR, new WolfDyeColorValueProcessor());
-        DataUtil.registerValueProcessor(Keys.DYE_COLOR, new SheepDyeColorValueProcessor());
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/data/DataRegistrar.java:770]: Where do these go now?

**Review by gabizou — CHANGES_REQUESTED** [review]
I think you've removed some processor registrations you didn't intend to remove.

File: src/main/java/org/spongepowered/common/data/DataRegistrar.java (hunk 1 above)
Comment (ImMorpheus) [reply src/main/java/org/spongepowered/common/data/DataRegistrar.java:770]: Aren't they registered in https://github.com/SpongePowered/SpongeCommon/blob/bleeding/src/main/java/org/spongepowered/common/data/util/DataUtil.java#L298 ?

Comment (gabizou) [reply src/main/java/org/spongepowered/common/data/DataRegistrar.java:770]: Ah yeah, subtle name change from value to data processor. good catch!

**Review by gabizou — APPROVED** [review]

File: src/main/java/org/spongepowered/common/data/DataRegistrar.java [hunk 2]
```diff
@@ -762,9 +770,7 @@ public static void setupSerialization() {
...
-        DataUtil.registerValueProcessor(Keys.DYE_COLOR, new ItemDyeColorValueProcessor());
+        DataUtil.registerValueProcessor(Keys.DYE_COLOR, new WolfDyeColorDataProcessor());
```
Comment (ImMorpheus) [inline src/main/java/org/spongepowered/common/data/DataRegistrar.java]: Is this registered twice ?

File: src/main/java/org/spongepowered/common/data/processor/data/item/ItemDyeColorDataProcessor.java [hunk 3]
```diff
@@ -22,33 +22,50 @@
...
+        final Block block = Block.getBlockFromItem(item);
+
+        return block.equals(Blocks.WOOL)
+                || block.equals(Blocks.STAINED_GLASS)
+                || block.equals(Blocks.CARPET)
+                || block.equals(Blocks.STAINED_GLASS_PANE)
+                || block.equals(Blocks.STAINED_HARDENED_CLAY)
+                || block.equals(Blocks.CONCRETE)
+                || block.equals(Blocks.CONCRETE_POWDER)
+                || block.equals(Blocks.BED);
```
Comment (liach) [inline src/main/java/org/spongepowered/common/data/processor/data/item/ItemDyeColorDataProcessor.java]: Are bed blocks and bed items paired? I suggest using `Items.BED` in checks.

Comment (JBYoshi) [reply src/main/java/org/spongepowered/common/data/processor/data/item/ItemDyeColorDataProcessor.java]: You are correct. Let me fix that.

File: src/main/java/org/spongepowered/common/data/DataRegistrar.java (hunk 2 above)
Comment (JBYoshi) [reply src/main/java/org/spongepowered/common/data/DataRegistrar.java]: It is. 😐

Comment (JBYoshi) [reply src/main/java/org/spongepowered/common/data/DataRegistrar.java]: Fixed.

**Comment by JBYoshi** [comment]
If there are no other comments, I'll merge this tonight.

**Pull request merged**